src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/53.diff.sql

```diff
@@ -0,0 +1,1223 @@
+            ( ResourceTypeId, ResourceSurrogateId, SearchParamId, SystemId1, Code1, CodeOverflow1, SingleValue2, LowValue2, HighValue2, SingleValue3, LowValue3, HighValue3, HasRange )
+    SELECT ResourceTypeId, ResourceSurrogateId, SearchParamId, SystemId1, Code1, CodeOverflow1, SingleValue2, LowValue2, HighValue2, SingleValue3, LowValue3, HighValue3, HasRange
+        FROM @TokenNumberNumberCompositeSearchParams
+    SET @AffectedRows += @@rowcount
+
+    IF @IsResourceChangeCaptureEnabled = 1 --If the resource change capture feature is enabled, to execute a stored procedure called CaptureResourceChanges to insert resource change data.
+    EXECUTE dbo.CaptureResourceIdsForChanges @Resources
+
+    IF @InitialTranCount = 0 COMMIT TRANSACTION
+
+    EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status=''End'',@Start=@st,@Rows=@AffectedRows
+END TRY
+BEGIN CATCH
+    IF @InitialTranCount = 0 AND @@trancount > 0 ROLLBACK TRANSACTION
+    IF error_number() = 1750 THROW -- Real error is before 1750, cannot trap in SQL.
+
+    EXECUTE dbo.LogEvent @Process=@SP,@Mode=@Mode,@Status=''Error'',@Start=@st;
+
+    IF @RaiseExceptionOnConflict = 1 AND error_number() = 2601 AND error_message() LIKE ''%''''dbo.Resource''''%version%''
+    THROW 50409, ''Resource has been recently updated or added, please compare the resource content in code for any duplicate updates'', 1;
+    ELSE
+    THROW
+END CATCH'
+            )
+
+            SET @logTime = getUTCdate()
+            EXECUTE dbo.LogEvent @Process='Drop Temp Types',@Status='Start',@Start=@logTime
+
+            DROP TYPE dbo.Tmp_NumberSearchParamList
+            DROP TYPE dbo.Tmp_QuantitySearchParamList
+            DROP TYPE dbo.Tmp_TokenNumberNumberCompositeSearchParamList
+            DROP TYPE dbo.Tmp_TokenQuantityCompositeSearchParamList
+
+            SET @logTime = getUTCdate()
+            EXECUTE dbo.LogEvent @Process='Drop Temp Types',@Status='End',@Start=@logTime
+
+            EXECUTE dbo.LogEvent @Process='Drop Orig Tables',@Status='Start',@Start=@logTime
+            drop table dbo.NumberSearchParam;
+            drop table dbo.QuantitySearchParam;
+            drop table dbo.TokenNumberNumberCompositeSearchParam;
+            drop table dbo.TokenQuantityCompositeSearchParam;
+
+            SET @logTime = getUTCdate()
+            EXECUTE dbo.LogEvent @Process='Rename Temp Tables',@Status='Start',@Start=@logTime
+            EXEC sp_rename 'Tmp_NumberSearchParam', 'NumberSearchParam';
+            EXEC sp_rename 'Tmp_QuantitySearchParam', 'QuantitySearchParam';
+            EXEC sp_rename 'Tmp_TokenNumberNumberCompositeSearchParam', 'TokenNumberNumberCompositeSearchParam';
+            EXEC sp_rename 'Tmp_TokenQuantityCompositeSearchParam', 'TokenQuantityCompositeSearchParam';
+        COMMIT TRANSACTION
+    END TRY
+    BEGIN CATCH
+        EXECUTE dbo.LogEvent @Process=@SP,@Status="Error",@Start=@st;
+        ROLLBACK TRANSACTION
+    END CATCH
```

Comment:
>END CATCH [](http://example.com/codeflow?start=4&length=9)

Add THROW as last statement in catch block

Reply:
Done